cluster/vagrant/util.sh

```diff
@@ -33,13 +33,62 @@ function detect-minions {
 # Verify prereqs on host machine  Also sets exports USING_KUBE_SCRIPTS=true so
 # that our Vagrantfile doesn't error out.
 function verify-prereqs {
-  for x in vagrant VBoxManage; do
+  for x in vagrant; do
     if ! which "$x" >/dev/null; then
       echo "Can't find $x in PATH, please fix and retry."
       exit 1
     fi
   done
 
+  local vagrant_plugins=$(vagrant plugin list | sed '-es% .*$%%' '-es%  *% %g' | tr ' ' $'\n')
+  local providers=(
+      # Format is:
+      #   provider_ctl_executable vagrant_provider_name vagrant_provider_plugin_re
+      # either provider_ctl_executable or vagrant_provider_plugin_re can
+      # be blank (i.e., '') if none is needed by Vagrant (see, e.g.,
+      # virtualbox entry)
+      vmrun vmware_fusion vagrant-vmware-fusion
+      vmrun vmware_workstation vagrant-vmware-workstation
+      prlctl parallels vagrant-parallels
+      VBoxManage virtualbox ''
+  )
+  local provider_found=''
+  local provider_bin
+  local provider_name
+  local provider_plugin_re
+
+  while [ "${#providers[@]}" -gt 0 ]; do
+    provider_bin=${providers[0]}
+    provider_name=${providers[1]}
+    provider_plugin_re=${providers[2]}
+    providers=("${providers[@]:3}")
+
+    # If the provider is explicitly set, look only for that provider
+    if [ -n "${VAGRANT_DEFAULT_PROVIDER:-}" ] \
+        && [ "${VAGRANT_DEFAULT_PROVIDER}" != "${provider_name}" ]; then
+      continue
+    fi
+
+    if ([ -z "${provider_bin}" ] \
+          || which "${provider_bin}" >/dev/null) \
```

Comment:
I get the following:

```shell
which: no vmrun in (/home/decarr/google-cloud-sdk/bin:/home/decarr/go/src/github.com/openshift/origin/_output/local/go/bin:/home/decarr/go/bin:/home/decarr/.rbenv/shims:/home/decarr/.rbenv/bin:/home/decarr/google-cloud-sdk/bin:/usr/lib64/qt-3.3/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/go/bin:/usr/local/sbin:/usr/sbin:/home/decarr/.local/bin:/home/decarr/bin:/home/decarr/.rvm/bin:/home/decarr/.rvm/bin)
which: no vmrun in (/home/decarr/google-cloud-sdk/bin:/home/decarr/go/src/github.com/openshift/origin/_output/local/go/bin:/home/decarr/go/bin:/home/decarr/.rbenv/shims:/home/decarr/.rbenv/bin:/home/decarr/google-cloud-sdk/bin:/usr/lib64/qt-3.3/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/go/bin:/usr/local/sbin:/usr/sbin:/home/decarr/.local/bin:/home/decarr/bin:/home/decarr/.rvm/bin:/home/decarr/.rvm/bin)
which: no prlctl in (/home/decarr/google-cloud-sdk/bin:/home/decarr/go/src/github.com/openshift/origin/_output/local/go/bin:/home/decarr/go/bin:/home/decarr/.rbenv/shims:/home/decarr/.rbenv/bin:/home/decarr/google-cloud-sdk/bin:/usr/lib64/qt-3.3/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/go/bin:/usr/local/sbin:/usr/sbin:/home/decarr/.local/bin:/home/decarr/bin:/home/decarr/.rvm/bin:/home/decarr/.rvm/bin)
```

can we change to do the following:

```
which "${provider_bin}" &>/dev/null
```

That is the last issue.